DoLah/Parser/TaskTokenizer.cpp

```diff
@@ -0,0 +1,77 @@
+#include "TaskTokenizer.h"
+
+
+namespace DoLah {
+    std::vector<std::string> TaskTokenizer::DEADLINE_INDICATOR = { "on", "by" };
+    std::string TaskTokenizer::SCHEDULE_INDICATOR = "from";
+    std::vector<std::string> TaskTokenizer::SCHEDULE_SEPARATOR = { "to", "until" };
+    std::string TaskTokenizer::tag = "#";
+
+    TaskTokenizer::TaskTokenizer() {
+    }
+
+
+    TaskTokenizer::~TaskTokenizer() {
+    }
+
+    std::vector<std::string> TaskTokenizer::findTags(std::vector<std::string> lineArr) {
+        std::vector<std::string> tags;
+        std::vector<std::string> moretags;
+
+        for (size_t i = 0; i < lineArr.size(); i++) {
+            moretags = ParserLibrary::explode(lineArr.at(i), tag);
+            tags.insert(tags.begin(), moretags.begin() + 1, moretags.end());
+        }
+
+        for (size_t i = 0; i < tags.size(); i++) {
+            tags.at(i) = ParserLibrary::tolowercase(tags.at(i));
+        }
+
+        tags = ParserLibrary::stringVectorUnique(tags);
+
+        return tags;
+    }
+
+    std::string TaskTokenizer::findDescription(std::vector<std::string> lineArr) {
+        return ParserLibrary::implode(lineArr, " ");
+    }
+
+    std::vector<std::tm> TaskTokenizer::findAndRemoveDate(std::vector<std::string> &lineArr) {
+        TaskTokenizer ct;
+        std::vector<std::tm> output;
+
+        try {
+            for (size_t i = lineArr.size() - 1; i > 0; i--) {
+                if (ParserLibrary::inStringArray(DEADLINE_INDICATOR, ParserLibrary::tolowercase(lineArr.at(i)))) {
+                    std::vector<std::string> subVec(lineArr.begin() + i + 1, lineArr.end());
+
+                    std::tm time = DateTimeParser::toDateFormat(subVec);
+
+                    output.push_back(time);
+                    lineArr.erase(lineArr.begin() + i, lineArr.end());
+                    return output;
```

Comment:
Personal Preference: I'm not a big fans of `return` in the middle of a function. It breaks the flow of the program execution.